src/test/java/org/broadinstitute/hellbender/tools/coveragemodel/cachemanager/ImmutableComputableGraphUnitTest.java

```diff
@@ -0,0 +1,81 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.cachemanager;
+
+import junit.framework.AssertionFailedError;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Unit tests for {@link ImmutableComputableGraph}
+ *
+ * TODO github/gatk-protected issue #803
```

Comment:
Probably worth splitting the github issue into different types of tests to write.